### PR TITLE
[Flink] Support analysis statistics in flink

### PR DESCRIPTION
**Pull request by herefree**
### Purpose
support statistics in flink

Linked issue: close #xxx

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        Snapshot snapshot = table.snapshotManager().latestSnapshot();
+        long mergedRecordCount = catalogTableStatistics.getRowCount();
+        long totalRecordCount = snapshot.totalRecordCount();
+        Preconditions.checkState(
+                totalRecordCount >= mergedRecordCount,
+                "totalRecordCount: $totalRecordCount should be greater or equal than mergedRecordCount: $mergedRecordCount.");
+        long totalSize =
+                table.newScan().plan().splits().stream()
+                        .flatMap(split -> ((DataSplit) split).dataFiles().stream())
+                        .mapToLong(dataFile -> dataFile.fileSize())
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:63]: Change "dataFile -> dataFile.fileSize()" to "DataFileMeta::fileSize".

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        // convert to paimon stats
+        return new Statistics(
+                snapshot.id(), snapshot.schemaId(), mergedRecordCount, mergedRecordSize);
+    }
+
+    /** Create Paimon statistics from given Flink columnStatistics. */
+    public static Statistics createTableColumnStats(
+            FileStoreTable table, CatalogColumnStatistics columnStatistics) {
+
+        Statistics statistics = table.statistics().get();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:77]: 'Optional.get()' without 'isPresent()' check

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        HashMap<String, ColStats<?>> tableColumnStatsMap = new HashMap<>();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:78]: Map<String, ColStats<?>> tableColumnStatsMap = new HashMap<>(fields.size());

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+            }
+            tableColumnStatsMap.put(
+                    field.name(), getPaimonColStats(field, catalogColumnStatisticsDataBase));
+        }
+        statistics.colStats().putAll(tableColumnStatsMap);
+        return statistics;
+    }
+
+    /** Convert Flink ColumnStats to Paimon ColStats according to Paimon column type. */
+    private static ColStats getPaimonColStats(
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:94]: ColStats<?>

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+                        null != stringColStat.getMaxLength() ? stringColStat.getMaxLength() : null);
+            }
+        } else if (colStat instanceof CatalogColumnStatisticsDataBoolean) {
+            CatalogColumnStatisticsDataBoolean booleanColStat =
+                    (CatalogColumnStatisticsDataBoolean) colStat;
+            if (typeRoot.equals(DataTypeRoot.BOOLEAN)) {
+                return ColStats.newColStats(
+                        field.id(),
+                        (booleanColStat.getFalseCount() > 0 ? 1L : 0)
+                                + (booleanColStat.getTrueCount() > 0 ? 1 : 0),
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:119]: (booleanColStat.getTrueCount() > 0 ? 1L : 0),

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+                                ? binaryColumnStatsData.getAvgLength().longValue()
+                                : null,
+                        null != binaryColumnStatsData.getMaxLength()
+                                ? binaryColumnStatsData.getMaxLength()
+                                : null);
+            }
+        }
+        throw new CatalogException(
+                String.format(
+                        "Flink does not support converting ColumnStats '%s' for Paimon column "
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/TableStatsUtil.java:281]: Change "converting" to "convert".

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                        + ", i INT"
+                        + ", l bigint"
+                        + ", PRIMARY KEY (id) NOT ENFORCED"
+                        + " ) WITH ("
+                        + " 'bucket' = '2'"
+                        + " )");
+        sql("INSERT INTO T VALUES ('1', 'a', 1, 1)");
+        sql("INSERT INTO T VALUES ('2', 'aaa', 1, 2)");
+        sql("ANALYZE TABLE T COMPUTE STATISTICS");
+        Statistics stats = paimonTable("T").statistics().get();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:50]: 'Optional.get()' without 'isPresent()' check

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+
+        Assertions.assertEquals(2L, stats.mergedRecordCount().getAsLong());
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:52]: 'OptionalLong.getAsLong()' without 'isPresent()' check

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        Assertions.assertTrue(stats.mergedRecordSize().isPresent());
+        Assertions.assertTrue(stats.colStats().isEmpty());
+    }
+
+    @Test
+    public void testAnalyzeTableColumn() throws Catalog.TableNotExistException {
+
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:59]: Delete the excess blank line.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        sql(
+                "INSERT INTO T VALUES ('3', 'bbbb', CAST('data' AS BYTES), 4, 19, 7.0, 1.0, 14.12345, true, cast(NULL as date), cast('2024-01-02 05:00:00' as TIMESTAMP_LTZ), CAST(NULL AS BINARY), CAST('example binary1' AS VARBINARY), 'aaa', 'aaa', CAST(NULL AS TINYINT), CAST(4 AS SMALLINT))");
+
+        sql(
+                "INSERT INTO T VALUES ('4', 'aa', CAST(NULL AS BYTES), 1, 1, 1.0, 1.0, 14.12345, false, cast(NULL as date), cast(NULL as TIMESTAMP_LTZ), CAST(NULL AS BINARY), CAST('example' AS VARBINARY), 'aba', 'aaab', CAST(NULL AS TINYINT), CAST(4 AS SMALLINT))");
+
+        sql("ANALYZE TABLE T COMPUTE STATISTICS FOR ALL COLUMNS");
+
+        Statistics stats = paimonTable("T").statistics().get();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:83]: 'Optional.get()' without 'isPresent()' check

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+        Assertions.assertEquals(4L, stats.mergedRecordCount().getAsLong());
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:85]: 'OptionalLong.getAsLong()' without 'isPresent()' check

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+                ColStats.newColStats(0, 4L, null, null, 0L, 1L, 1L), colStats.get("id"));
+        Assertions.assertEquals(
+                ColStats.newColStats(1, 4L, null, null, 0L, 2L, 4L), colStats.get("name"));
+
+        Assertions.assertEquals(
+                ColStats.newColStats(2, null, null, null, 1L, null, null),
+                colStats.get("bytes_col"));
+
+        Assertions.assertEquals(
+                ColStats.newColStats(3, 2L, new Integer(1), new Integer(4), 0L, null, null),
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkAnalyzeTableITCase.java:98]: Why add two unnecessary boxing here?

**Comment by herefree** [comment]
@wwj6591812 thanks for your review,I have completed the modification.

**Pull request merged**